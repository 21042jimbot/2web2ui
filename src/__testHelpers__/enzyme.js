export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
