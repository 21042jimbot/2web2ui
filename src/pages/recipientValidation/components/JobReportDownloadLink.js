import React from 'react';
import { Button } from '@sparkpost/matchbox';
import { FileDownload } from '@sparkpost/matchbox-icons';
import DownloadLink from 'src/components/downloadLink/DownloadLink';

const JobReportDownloadLink = ({ complete, rejectedUrl, uploadedFile }) => {
  if (!complete) {
    return null;
  }

  return (
    <DownloadLink
      component={Button}
      to={rejectedUrl ? rejectedUrl : uploadedFile}
    >
      <span>Download Results&nbsp;</span>
      <FileDownload />
    </DownloadLink>
  );
};

export default JobReportDownloadLink;
