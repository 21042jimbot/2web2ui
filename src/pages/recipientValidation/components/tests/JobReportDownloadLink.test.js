import React from 'react';
import { shallow } from 'enzyme';
import JobReportDownloadLink from '../JobReportDownloadLink';

describe('JobReportDownloadLink', () => {
  const subject = (props = {}) => shallow(
    <JobReportDownloadLink {...props} />
  );

  it('renders null when no file is provided', () => {
    expect(subject()).toBeEmptyRender();
  });

  it('renders a download link', () => {
    const wrapper = subject({ href: 'http://example.com/rejected.csv', status: 'success' });
    expect(wrapper).toMatchSnapshot();
  });
});
