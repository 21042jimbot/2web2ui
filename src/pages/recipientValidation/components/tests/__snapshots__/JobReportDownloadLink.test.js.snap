// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobReportDownloadLink renders a download link 1`] = `
<DownloadLink
  component={[Function]}
  to="http://example.com/upload.csv"
>
  <span>
    Download ResultsÂ 
  </span>
  <FileDownload />
</DownloadLink>
`;
